### PR TITLE
[GSK-1201] Enable testing search bar

### PR DESCRIPTION
**Pull request by henchaves**
## Description

This PR aims to create a search bar on Testing tab, allowing users to filter Test suites by name.

## Related Issue

[GSK-1201 (available on Linear)](https://linear.app/giskard/issue/GSK-1201/enable-testing-search-bar)

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

### CODE REVIEW

File: frontend/src/views/main/project/TestSuites.vue [hunk 1]
```diff
@@ -177,14 +181,19 @@ onActivated(async () => {
...
-function renameSuite(suiteId: number, name: string) {
+async function renameSuite(suiteId: number, name: string) {
     const currentSuite = testSuitesStore.testSuites.find(s => s.id === suiteId);
 
     if (currentSuite) {
         currentSuite.name = name;
-        testSuitesStore.updateSuiteName(currentSuite);
-
-        mixpanel.track('Renamed test suite', {suiteId});
+        await testSuitesStore.updateSuiteName({
```
Comment (andreybavt) [inline frontend/src/views/main/project/TestSuites.vue:189]: why do you have to create a new object instead of sending a `currentSuite` with updated name?

Comment (henchaves) [reply frontend/src/views/main/project/TestSuites.vue:189]: For some reason, it doesn't update the name on the list after sending currentSuite. This was the alternative I found to make automatically update.

Comment (henchaves) [reply frontend/src/views/main/project/TestSuites.vue:189]: I'm gonna test if this behaviour happens again if I use `currentSuite`

Comment (henchaves) [reply frontend/src/views/main/project/TestSuites.vue]: Now it's working with `currentSuite`. Just pushed it.

**Review by andreybavt — APPROVED** [review]

**Pull request merged**